MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm

```diff
@@ -611,7 +611,7 @@
 												   const VkDescriptorSetLayout* pSetLayouts,
 												   VkDescriptorSet* pDescriptorSets) {
 	if (_allocatedSetCount + count > _maxSets) {
-		return mvkNotifyErrorWithText(VK_ERROR_INITIALIZATION_FAILED, "The maximum number of descriptor sets that can be allocated by this descriptor pool is %d.", _maxSets);
+		return mvkNotifyErrorWithText(VK_ERROR_OUT_OF_POOL_MEMORY_KHR, "The maximum number of descriptor sets that can be allocated by this descriptor pool is %d.", _maxSets);
```

Comment:
I think the behavior here needs to be dependent on whether the app enabled VK_KHR_maintenance1 in vkCreateDevice.  The old behavior is correct when its disabled.  In VK_KHR_maintenance1, I don't think it should do mvkNotifyErrorWithText even, just return VK_ERROR_OUT_OF_POOL_MEMORY_KHR.  It's normal for applications to expect a failure here.

Reply:
@danginsburg 

The logging of the `VK_ERROR_OUT_OF_POOL_MEMORY_KHR` result is consistent with that `VkResult` being labelled as an error (`VK_ERROR_...`). Are you recommending that for this one error result, we consider it to be a non-error `VkResult` (like say `VK_INCOMPLETE`)...and not log it?

Reply:
I would prefer that, yes.  Dota 2 relies on this failure and checks for the error.  Logging it produces spam which I don't think needs to be there.